app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -257,80 +275,104 @@ public void cameraPerm (){
                 dateTime.get(Calendar.MONTH), dateTime.get(Calendar.DAY_OF_MONTH)).show();
     }
 
 
     private void submitMood() throws DescriptionTooLongException {
-        if(editflag) {
-            Mood mood = editmood;
-            ElasticSearchController.EditMoodTask editMoodTask = new ElasticSearchController.EditMoodTask(this);
-            if (showPublicOn) {
-                mood.setPublic();
-            } else {
-                mood.setPrivate();
-            }
-            if (locationOn) {
-                mood.setMapPosition(latitude, longitude);
-                // this is the setter for latitude and longitude
-            }
-            else {
-                mood.setNullLocation();
-            }
-            mood.setEmotionalState(emotionalState);
-            mood.setSocialSit(socialSit);
-            mood.setDescription(String.valueOf(inputMoodDescription.getText())); //TODO: append the emotionalState upon fetching from Elasticsearch
-            mood.setDate(dateTime.getTime());
+        try {
+            if (editflag) {
+                Mood mood = editmood;
+                ElasticSearchController.EditMoodTask editMoodTask = new ElasticSearchController.EditMoodTask(this);
+                if (showPublicOn) {
+                    mood.setPublic();
+                } else {
+                    mood.setPrivate();
+                }
+                if (locationOn) {
+                    mood.setMapPosition(latitude, longitude);
+                    // this is the setter for latitude and longitude
+                } else {
+                    mood.setNullLocation();
+                }
+                if (emotionalState.equals("")) {
+                    throw new RuntimeException();
+                } else {
+                    mood.setEmotionalState(emotionalState);
+                }
 
-            if(encodedPhoto != null) {
-                mood.setImage(encodedPhoto);
-            }
-            else {
-                mood.setNullImage();
-            }
+                mood.setEmoji(emojiID);
 
-            editMoodTask.execute(mood);
+                mood.setSocialSit(socialSit);
 
-            Log.d("tag", "Editing mood");
-        }
-        else {
-            ElasticSearchController.AddMoodTask addMoodTask = new ElasticSearchController.AddMoodTask(this);
-            Participant participant = FeelTripApplication.getParticipant();
-            Mood mood = new Mood(participant.getUserName());
-            if (showPublicOn) {
-                mood.setPublic();
+                if (inputMoodDescription.getText().toString().equals("")) {
+                    throw new NullPointerException();
+                } else {
+                    mood.setDescription(String.valueOf(inputMoodDescription.getText()));
+                }
+
+                mood.setDate(dateTime.getTime());
+
+                if (encodedPhoto != null) {
+                    mood.setImage(encodedPhoto);
+                } else {
```

Comment:
Ye, this segment is here for the sole purpose of editing an existing mood with an image, since we need some way of loading its mood into memory.